### PR TITLE
fix(build_cluster_test): Use remote_scylla_yaml to edit the yaml file

### PR DESCRIPTION
**Pull request by igorsimb**
ScyllaYaml put "none" as listen_address and rpc_address. This fixes the issue. Also, increased test duration to prevent unwanted timeouts.

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by igorsimb** [comment]
Job: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/igorsimb/job/features-dns-cluster-5min-test_job_2/
"Send email with result" is failing, I don't know why. I noticed my personal github email in the logs, it may have something to do with it.

**Review by ShlomiBalalis — DISMISSED** [review]
(Other than the mail thing) LGTM
Though that test duration is oddly specific

**Comment by fruch** [comment]
@igorsimb 

The mail is not sent, cause no results file
```
Results file not found
```

I'm guessing this old test class doesn't have an email reporter like Longevity and others.

Need to implement an email reporter for all functional tests

File: test-cases/features/dns-cluster-5min.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # TODO: need to qualify
 
-test_duration: 20
+test_duration: 77
```
Comment (vponomaryov) [inline test-cases/features/dns-cluster-5min.yaml:3]: Place this and another timeout change to a separate commit because both are unrelated to the main goal of the main change here.

**Review by vponomaryov — CHANGES_REQUESTED** [review]

File: build_cluster_test.py [hunk 2]
```diff
@@ -70,13 +70,15 @@ def test_use_public_dns_names(self):
...
 
         for node in self.db_cluster.nodes:
-            node.remoter.run('sudo sed -i.bak s/{0}/{1}/g /etc/scylla/scylla.yaml'.format(node.private_ip_address,
-                                                                                          addresses.get(
-                                                                                              node.private_ip_address,
-                                                                                              "")))
-            for seed in seeds:
-                node.remoter.run(
-                    'sudo sed -i.bak2 s/{0}/{1}/g /etc/scylla/scylla.yaml'.format(seed, addresses.get(seed, "")))
+            with node.remote_scylla_yaml() as scylla_yml:
```
[the remark this answers is not in the document]
Comment (igorsimb) [reply build_cluster_test.py:73]: Done

File: jenkins-pipelines/features-dns-cluster-5min.jenkinsfile [hunk 3]
```diff
@@ -9,5 +9,5 @@ longevityPipeline(
     test_name: 'build_cluster_test.BuildClusterTest.test_use_public_dns_names',
     test_config: 'test-cases/features/dns-cluster-5min.yaml',
 
-    timeout: [time: 60, unit: 'MINUTES']
+timeout: [time: 60, unit: 'MINUTES']
```
Comment (vponomaryov) [inline jenkins-pipelines/features-dns-cluster-5min.jenkinsfile]: Mistaken change.

**Review by vponomaryov — CHANGES_REQUESTED** [review]
One mistaken change in the main commit. Plus, the main commit description is not updated.

File: jenkins-pipelines/features-dns-cluster-5min.jenkinsfile (hunk 3 above)
Comment (igorsimb) [reply jenkins-pipelines/features-dns-cluster-5min.jenkinsfile]: Corrected

**Review by vponomaryov — DISMISSED** [review]

File: jenkins-pipelines/features-dns-cluster-5min.jenkinsfile (hunk 3 above)
Comment (vponomaryov) [reply jenkins-pipelines/features-dns-cluster-5min.jenkinsfile]: The only change that will be useful here is deletion of the line 12 with the "timeout" parameter.
Why? Because it is not used.
Proof is in the `vars/longevityPipeline.groovy` file which gets called here.

**Comment by fruch** [comment]
@igorsimb 

```timeout: [time: 60, unit: 'MINUTES']``` from `features-dns-cluster-5min.jenkinsfile` 
this parameter isn't in use anymore by longevity pipelines (the Jenkins timeouts are calculated from the `test_duration`)

Lots of pipeline files still has those, since no one went over all and removed. but here once you are touching specific test case, it's a good time to get rid of it. (to avoid confusion, of where to change a case duration/timeout)

**Comment by roydahan** [comment]
> @igorsimb
> 
> `timeout: [time: 60, unit: 'MINUTES']` from `features-dns-cluster-5min.jenkinsfile` this parameter isn't in use anymore by longevity pipelines (the Jenkins timeouts are calculated from the `test_duration`)
> 
> Lots of pipeline files still has those, since no one went over all and removed. but here once you are touching specific test case, it's a good time to get rid of it. (to avoid confusion, of where to change a case duration/timeout)

We should either stop commenting about this line in PRs or one to just remove it from all pipelines.

**Review by fruch — APPROVED** [review]
LGTM
